### PR TITLE
add v0 tech spec

### PR DESCRIPTION
**Pull request by ziyangczi**
Added v0 tech spec after clean up

### CODE REVIEW

File: docs/v0-tech-spec.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+  * This will help reduce regressions in accessibility, performance, and SEO
+  * Highlight improvements in aforementioned areas
+
+#### Backend testing
+* API functional testing
+  * Unit test for api setup for different endpoints
+* API Load testing 
+  * Confirm traffic we can handle in the current setup
+* Health test
+  * Setup health check for the http server
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:174]:
```suggestion
  * Health check for the http server
```

File: docs/v0-tech-spec.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+### Release Plan
+First version of the site will be released on production by June 30, 2021.
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:177]:
```suggestion
First version of the site was released on production on June 30, 2021. Subsequent major and minor versions to follow.
```

File: docs/v0-tech-spec.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+#### Domain name
+Decision: We will use napari-hub.org with naparihub.org redirecting to napari-hub.org
+Need to work with infra team to purchase domain or use subdomain subpage of napari.org.
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:181]:
```suggestion
```

File: docs/v0-tech-spec.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+#### Hosting strategy
+* We will use CZI hosted AWS since it will allow us more flexibility, is more future proof than Github pages and we 
+  have a budget for services. We can take advantage of AWS services like Lambda and load balancing.
+* We will work with CZI Infra Team to understand the components they have available for use. Terraform will be used. 
+  Option to open source the Terraform files after release is still an open question on how best to do this.
+
+#### Staging site
+* We should have a separate staging site for testing
+* We will likely put up a coming soon page on the production site until release
+Solutions
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:192]:
```suggestion

### Solutions
```

File: docs/v0-tech-spec.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+ 
+#### API
+Deploy AWS lambda (to avoid machine level maintenance) to support API where we parse pypi result with a frequency 
+of 5min (configurable, fine tune later between 1min to 1 hour), and cache to aws s3. We will use the public google 
+bigquery dataset as a backup data source when the main query fails (this should also triggers an alert to our team
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:197]:
```suggestion
bigquery dataset as a backup data source when the main query fails (this should also trigger an alert to our team 
```

File: docs/v0-tech-spec.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+* We should have a separate staging site for testing
+* We will likely put up a coming soon page on the production site until release
+Solutions
+ 
+#### API
+Deploy AWS lambda (to avoid machine level maintenance) to support API where we parse pypi result with a frequency 
+of 5min (configurable, fine tune later between 1min to 1 hour), and cache to aws s3. We will use the public google 
+bigquery dataset as a backup data source when the main query fails (this should also triggers an alert to our team 
+channel to indicate that the lambdas have switched to the backup approach, eng team is responsible to react to such 
+alerts and make adjustment (be it pypi outage or pypi website structure change  to address the failure and bring the
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:199]:
```suggestion
alerts and make adjustment (be it pypi outage or pypi website structure change to address the failure and bring the
```

File: docs/v0-tech-spec.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+parsing back online). The API to be built should return a list of plugin names, in the format of 
+“[“plugin_1”, “plugin_2]”, where each individual plugin name can then be used to query qypi api
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:201]:
```suggestion
“[“plugin_1”, “plugin_2]”, where each individual plugin name can then be used to query pypi API.
```

File: docs/v0-tech-spec.md [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+ 
+##### endpoints
+* /plugins
+  * return list of plugin and versions:
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:205]:
```suggestion
  * return list of plugin and versions
```

File: docs/v0-tech-spec.md [hunk 9]
```diff
@@ -0,0 +1,271 @@
...
+* /plugins/index
+  * return all plugins’ metadata used in main page
+* /plugins/{plugin_name}
+  * return metadata for a plugin
+* /plugins/{plugin_name}/version/{version}: 
+  * return metadata for a specific version of plugin
+
+#### Overall back-end technology:
+We decided to use AWS which we will use services like Lambda, S3 and API Gateway.
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:214]:
```suggestion
We decided to use AWS which we will use services like Lambda, S3 and API Gateway. A software architecture diagram can be found [here](https://lucid.app/lucidchart/1ad7e89e-7265-4a04-b37b-135e7a57432c/view).
```

File: docs/v0-tech-spec.md [hunk 10]
```diff
@@ -0,0 +1,271 @@
...
+ 
+#### Overall front-end technology:
+After building some proof of concepts and testing, we decided to use React + Next.js and TailwindCSS. Further 
+information to support this decision can be found in this document:Next.js vs SvelteKit
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:218]: We may want to publish a summary of the findings. If we're not right now, let's remove this line.

File: docs/v0-tech-spec.md [hunk 11]
```diff
@@ -0,0 +1,271 @@
...
+    support potentially 1,000s of plugins in future versions.
+  * We will use Plausible for analytics for the first version. In future releases, we may need more detailed analytics 
+    and should provide this without the use of a cookie banner.
+* Impact on other teams/projects
+  * Depending on our solution to getting the plugin list data, this could be something that the napari desktop 
+    application could use as well. This may help to keep consistent.
+  * After starting the implementation, we learned that there is a CZI Science effort to create a unified design system 
+    based on Material UI. After doing a spike to figure out how much effort and the value we would get from using this 
+    framework, we decided to integrate it into our implementation. Research from the spike can be found in this 
+    document Material UI Integration
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:232]: Same as above, if we're not going to publish the findings, let's remove this line.

File: docs/v0-tech-spec.md [hunk 12]
```diff
@@ -0,0 +1,271 @@
...
+* User support
+* Who is responsible for the maintenance of this site?
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:234]:
```suggestion
For feature requests, please file [a new discussion under the ideas category](https://github.com/chanzuckerberg/napari-hub/discussions/categories/ideas).
For bug reports, please file [a new issue](https://github.com/chanzuckerberg/napari-hub/issues).
```

File: docs/v0-tech-spec.md [hunk 13]
```diff
@@ -0,0 +1,271 @@
...
+ 
+ 
+## Success Evaluation
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:237]:
```suggestion
```

File: docs/v0-tech-spec.md [hunk 14]
```diff
@@ -0,0 +1,271 @@
...
+Achieve OKR “At least 25 plugins discoverable through a plugin discovery site” by end of H1 2021
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:238]:
```suggestion
```

File: docs/v0-tech-spec.md [hunk 15]
```diff
@@ -0,0 +1,271 @@
...
+* Milestone April 5th: Plugin Page Design Complete and Ready to Build
+* Milestone May 6th: Discovery Page Design Complete and Ready to Build
+* Milestone June 30th: plugin discovery site alpha/v0 release
+
+ 
+### Future work
+* First version will be minimal
+* We will iteratively add new features and get feedback from users in a continuous cycle of future releases
+ 
+## Questions
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:250]: Please remove this entire section.

File: docs/v0-tech-spec.md [hunk 16]
```diff
@@ -0,0 +1,271 @@
...
+  * Search form fields:
+    * Global search [input/textbox] - Spreadsheet with fields and info defined. We will use all items from the “Search” 
+      column. More details for this feature defined under Features: Global Search Bar.
+    * Filter [checkboxes] - Spreadsheet with fields and info defined. We will use all items from the “Filter” column. 
+      More details for this feature defined under Features: Filter/Sort
+    * Sort [dropdown/radio buttons] - Spreadsheet with fields and info defined. We will use all items from the “Sort” 
+      column. More details for this feature defined under Features: Filter/Sort
+  * Plugin list data to display
+    * Fields and info specification on Github. We will use all items from the “List view” column
+    * Link to details page
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:29]:
```suggestion
    * Link to details page
    * We will add an “exclude” list for plugins to exclude from the list. The exclusions will be stored in a file and plugin data returned will not include those excluded plugins.
```

File: docs/v0-tech-spec.md [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+* /plugins/{plugin_name}/version/{version}: 
+  * return metadata for a specific version of plugin
+
+#### Overall back-end technology:
+We decided to use AWS which we will use services like Lambda, S3 and API Gateway.
+ 
+#### Overall front-end technology:
+After building some proof of concepts and testing, we decided to use React + Next.js and TailwindCSS. Further 
+information to support this decision can be found in this document:Next.js vs SvelteKit
+
```
Comment (justinelarsen) [inline docs/v0-tech-spec.md:219]:
```suggestion

## Publishing of plugins
* To be transparent and not require an approval process as well as share the responsibility to flagging bad plugins, we will publish activity to the hub-updates stream on the [napari Zulip](https://napari.zulipchat.com). Activity will be published for brand new plugins, new versions of plugins as well as plugin removals.

```

**Review by justinelarsen — CHANGES_REQUESTED** [review]
Thanks for this PR @ziyangczi ! Overall, it looks good. I suggested a bunch of edits.

File: docs/v0-tech-spec.md [hunk 18]
```diff
@@ -0,0 +1,271 @@
...
+### Features
+#### Global Search Bar
+##### Technical decision
+Filter on the client-side using a JS indexing library like fuse.js, lunr or minisearch.
+Pros: Quick search for medium sized data and provides more complex search capabilities like fuzzy search out of the box.
+Cons: Requires developer time to get familiar with the libraries and how to implement them. Libraries are fixed in what 
+they can support, but they should be good enough for our needs.
+
+##### Technical requirements
+* TBD Specifications on ordering, fuzzy search, etc. will be determined once Jeremy researches options in Solution #3.
```
[the remark this answers is not in the document]
Comment (codemonkey800) [reply docs/v0-tech-spec.md]: Added! 🔥

**Review by justinelarsen — APPROVED** [review]
Looks great to me! Thanks for making all my suggested changes! 😄 🎉

File: docs/v0-tech-spec.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (justinelarsen) [reply docs/v0-tech-spec.md]: Thank you! 😃

**Pull request merged**